oci/spec_opts.go

```diff
@@ -1278,3 +1286,19 @@ func WithDevShmSize(kb int64) SpecOpts {
 		return ErrNoShmMount
 	}
 }
+
+// updateSingleMountWithReadonly is used by the options which are trying to
+// get user/group information from container's rootfs. Since the option does
+// read operation only, this helper will append ReadOnly mount option to
+// prevent linux kernel from syncing whole filesystem.
+//
+// FIXME(fuweid):
+//
+// Currently, it only works for overlayfs. I think we can apply it to other
+// kinds of filesystem.
+func updateSingleMountWithReadonly(mounts []mount.Mount) []mount.Mount {
```

Comment:
Can we get the image / parent snapshot as an parameter and `View` it temporarily instead?

Reply:
`View` will create new snapshot. It is not necessary. I added TODO in that code that `Mount` API might support RO option.